packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/JsonMrwSerializationTypeProviderTests.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.Input;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Tests
+{
+    internal class JsonMrwSerializationTypeProviderTests
```

Comment:
Given the feedback from @ArcturusZhang lets rename this MrwSerializationTypeProvider since mrw handles both xml and json

Reply:
I see if the multiple serializations are handling different frameworks, then I am fine with multiple type providers.
My initial point was that since we are always implementing the same interface (for MRW as example) we should have different providers.
And I think in the case of enums, it should returns the array with only one element (in fixed case) or nothing (in extensible case)